### PR TITLE
PIM implementation of transitive closure

### PR DESCRIPTION
**Pull request by hugoabbot**
*Algorithm needs further reviewing to ensure that it's optimal and valid. As discussed in weekly meeting, this benchmark might be added before the final submission deadline if time permits. 

*Needs to be moved from PIMbench to misc-bench

## Transitive Closure

Benchmark uses the Floyd-Warhsall algorithm to perform transitive closure on a directed and weighted graph. `loadVector()` is not used in the timing as its functionality is possible using just PIM commands, but is not currently supported by the current API.

## Input

Specified input is supported given that the format is valid and supported. Examples are given in `datafiles/`. Additionally, the executable can be run without this specification, which defaults to a matrix with `256` nodes and a sparsity rate of `50%`.

### CODE REVIEW

File: PIMbench/cpp-transitive-closure/Makefile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Makefile: C++ version of transitive closure
+# Copyright 2024 LavaLab @ University of Virginia. All rights reserved.
```
Comment (deyuan) [inline PIMbench/cpp-transitive-closure/Makefile:2]: This copyright info needs to be updated

File: PIMbench/cpp-transitive-closure/Makefile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+include ../Makefile.common
+
+EXEC := transitive-closure.out
+SRC := transitive-closure.cpp
+
+FOPENMP := -fopenmp
```
Comment (deyuan) [inline PIMbench/cpp-transitive-closure/Makefile:9]: Need to guard this with USE_OPENMP flag. Example: cpp-gemm/Makefile

**Comment by fasiddique** [comment]
I would like to keep this on hold for now. If time permits we can add it to the paper.

**Review by fasiddique — APPROVED** [review]
Thank you!

**Pull request merged**